inc/tasks/promotions.php

```diff
@@ -237,6 +259,15 @@ function task_promotions($task)
 			if($usergroup_select == "usergroup")
 			{
 				$db->update_query("users", array('usergroup' => $promotion['newusergroup']), "uid IN(".implode(",", $uids).")");
+
+				if(!empty($updateDisplayGroups))
+				{
+					$userIDs = implode(',', $updateDisplayGroups);
+
+					$db->update_query('users', array('displaygroup' => 0), "uid IN ({$userIDs})");
+
+					$updateDisplayGroups = [];
```

Comment:
Use `$updateDisplayGroups = array();`

Reply:
Why this is required to reset?

Reply:
> Why this is required to reset?

Because `format_name` does not check for the actual membership of the user to the `displaygroup`. So users might be shown how _belonging_ to any group regardless of that being the case.

I think this is a bug, not a feature.

My suggestions should be tested, tho, as I did not test them myself.

Reply:
Thx!